### DISCUSSION TITLE
Why is the row not full

### DISCUSSION THREAD

**Discussion by laowus** [flet-dev/flet#2569] [Q&A]
`from flet import *

def main(page: Page):
    page.padding = 0
    page.scroll

    def getImgSrc(i):
        return f"https://picsum.photos/200/200?{i}"

    grs = Row(
        wrap=True,
        scroll=ScrollMode.ALWAYS,
        expand=1,
        run_spacing=30,
    )

    for i in range(1, 200):
        grs.controls.append(
            Container(
                content=Row(
                    [
                        Image(
                            src=getImgSrc(i),
                            width=40,
                            height=40,
                            fit="cover",
                            border_radius=10,
                        ),
                        Text(
                            f"title{i}",
                            width=180,
                            height=20,
                            weight="bold",
                            color=colors.WHITE,
                        ),
                        Text(
                            f"writer{i}",
                            width=80,
                            height=20,
                            no_wrap=True,
                            weight="bold",
                            color=colors.WHITE,
                        ),
                    ]
                ),
                bgcolor=colors.BLACK,
                padding=5,
                opacity=0.7,
                animate=300,
                border_radius=10,
                border=border.all(width=1),
                width=320,
                margin=margin.only(right=60, left=10),
            ),
        )

    page.add(grs)

app(target=main)

`

[
![image](https://github.com/flet-dev/flet/assets/11734379/374d1e67-f544-4d0a-8625-d0561274015d)
](url)

**Comment by AngaWende** [comment]
in the code the width of the row is the sum of the containers in each line
setting the property width with the size of the window_width or higher worked here

grs = Row(
        width=page.window_width,
        wrap=True,
        scroll=ScrollMode.ALWAYS,
        expand=1,
        run_spacing=30,
    )

[Gravação de tela de 2024-02-06 18-33-30.webm](https://github.com/flet-dev/flet/assets/106128009/787f7a59-e794-4d71-9431-7448f5256df1)

**Reply by laowus** [reply]
thanks

**Reply by laowus** [reply]
if row(wrap) change gridview? 
The height and width of sub elements cannot be set, it seems to be set by the gridview


`from flet import *


def main(page: Page):
    page.padding = 0

    def page_resize(e):
        # print("New page size:", page.window_width, page.window_height)
        grs.width = page.window_width
        page.update()

    page.on_resize = page_resize

    def getImgSrc(i):
        return f"https://picsum.photos/200/200?{i}"

    grs = GridView(
        expand=1,
        run_spacing=30,
        runs_count=3,
        child_aspect_ratio=5,
        width=page.window_width,
    )

    for i in range(1, 200):
        grs.controls.append(
            Container(
                content=Row(
                    [
                        Image(
                            src=getImgSrc(i),
                            width=40,
                            height=40,
                            fit="cover",
                            border_radius=10,
                        ),
                        Text(
                            f"title{i}",
                            width=180,
                            height=20,
                            weight="bold",
                            color=colors.WHITE,
                        ),
                        Text(
                            f"writer{i}",
                            width=80,
                            height=20,
                            no_wrap=True,
                            weight="bold",
                            color=colors.WHITE,
                        ),
                    ]
                ),
                bgcolor=colors.BLACK,
                padding=5,
                opacity=0.7,
                animate=300,
                border_radius=10,
                border=border.all(width=1),
                width=300,
                margin=margin.only(right=60, left=10),
            ),
        )

    page.add(grs)


app(target=main)
`
[https://raw.githubusercontent.com/huangxulei/flet-song/main/views/gr.png](url)

**Comment by AngaWende** [comment]
in this case, it looks better to set the size of containers with the properties of the gridview and it fits automatically at the resize

    from flet import *
    
    def main(page: Page):
    
        def getImgSrc(i):
            return f"https://picsum.photos/200/200?{i}"
    
        grs = GridView(
            expand=1,
            run_spacing=30,
            child_aspect_ratio=8,
            max_extent=page.window_width // 3,
        )
    
        for i in range(1, 50):
            grs.controls.append(
                Container(
                    content=Row(alignment=MainAxisAlignment.SPACE_AROUND,
                                controls=[ ... ]
                                ),
                    bgcolor=colors.BLACK,
                    padding=5,
                    opacity=0.7,
                    border_radius=10,
                ),
            )
        page.add(grs)
    
    app(main)

**Reply by laowus** [reply]
Thank you very much. Perhaps I need to further understand gridview
